lib/sensors/led_driver.cpp

```diff
@@ -0,0 +1,97 @@
+#include "led_driver.hpp"
+
+namespace hyped::sensors {
+
+std::optional<LedDriver> LedDriver::create(core::ILogger &logger,
+                                           std::shared_ptr<io::II2c> i2c,
+                                           const std::uint8_t device_address)
+{
+  if (device_address != kDefaultLedDriverAddress) {
+    logger.log(core::LogLevel::kFatal, "Invalid device address for LED driver");
+    return std::nullopt;
+  }
+
```

Comment:
avoid using whitespace (see the style guide [here](https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace)). the metaphor is that whitespace in code is similar to whitespace in an essay, it should separate paragraphs but you wouldn't have a line break after every sentence